web_services/foxx/brightid4/tests/verifications.js

```diff
@@ -61,26 +67,35 @@ describe('verifications', function () {
     contextsColl.truncate();
     sponsorshipsColl.truncate();
   });
-  it('should be able to map only a single contextId to each user', function() {
-    db.linkContextId('1', 'testContext', 'used', 5);
-    db.linkContextId('2', 'testContext', 'old', 15);
-    db.linkContextId('2', 'testContext', 'new', 25);
-  });
-  context('linkContextId()', function(){
-    it('should throw "contextId is duplicate" for not duplicate contextId', function(){
+  context('linkContextId()', function() {
+    it('should not be able to link contextId to not verified user', function() {
+      (() => {
+        db.linkContextId('1', 'testContext', 'used', 5);
+      }).should.throw('user is not verified for this context');
+    });
+    it('should throw "contextId is duplicate" for used contextId', function(){
+      db.linkContextId('2', 'testContext', 'used', 5);
       (() => {
-        db.linkContextId('3', 'testContext', 'used', 30);
+        db.linkContextId('3', 'testContext', 'used', 10);
       }).should.throw('contextId is duplicate');
     });
     it('should return add link if contextId and timestamp are OK', function(){
-      db.linkContextId('3', 'testContext', 'testContextId', 30);
+      db.linkContextId('3', 'testContext', 'testContextId', 10);
       db.getUserByContextId(contextIdsColl, 'testContextId').should.equal('3');
     });
+    it('should not be able to link new contextId before 8 hours', function(){
+      (() => {
+        db.linkContextId('3', 'testContext', 'testContextId2', 15);
+      }).should.throw('one contextId can be linked each 8 hours');
+    });
+    it('should be able to link new contextId after 8 hours', function(){
+      db.linkContextId('3', 'testContext', 'testContextId2', 8*3600*1000 + 15);
+    });
```

Comment:
This test will need to be fixed to check for 3 links in 24 hours

Reply:
This test is updated for 3 links every day